### ISSUE TITLE
开启Netflix分流服务器以后，对Emby，Plex的影响

### ISSUE THREAD

**Issue by jackiezhang29** [fw876/helloworld#421]
家里有群晖，除了日常功能以外，还有PT下载、Plex服务器、Emby服务器。已经把群晖的内网IP设置成了不走代理流量，避免PT下载误伤VPS流量，但是在开启Netflix分流服务器以后，Emby和Plex获得的广域网IP是VPS的IP地址，造成外网无法访问Plex服务器。
自己找过解决办法，就是关闭Netflix分流服务器，一切正常。也尝试在Netflix分流的IP里没有找到和Plex相关的域名和IP，不知道为何只要开启Netflix分流服务器，Plex和Emby就无法获取本地公网IP。

**Comment by qiqiandfei** [comment]
> 家里有群晖，除了日常功能以外，还有PT下载，Plex服务器，Emby服务器。已经把群晖的内网IP设置变成了不走代理流量，避免PT下载误伤VPS流量，但是在开启Netflix分流服务器以后，Emby和Plex获得的广域网IP是VPS的IP地址，造成外部网无法访问Plex服务器。
> 自己找过解决办法，就是关闭Netflix分流服务器，一切正常。也尝试在Netflix分流的IP里没有找到和Plex相关的域名和IP，不知道为何只要开启Netflix分流服务器，Plex和Emby就无法获取本地公网IP。

我觉得这不是问题，代理服务都是正向的，也就是你群晖常用端口的出口流量才会走代理（ssrp客户端页面可以设置‘仅常用端口（不走p2p流量到代理）’），所以PT上传\下载不会误伤VPS流量。你群晖获取的ip地址是VPS地址这也是正常的，因为一般都是curl请求获取公网地址，走的80/443端口属于常用端口，常用端口默认走代理了～

**Comment by jackiezhang29** [comment]
> > 家里有群晖，除了日常功能以外，还有PT下载，Plex服务器，Emby服务器。已经把群晖的内网IP设置变成了不走代理流量，避免PT下载误伤VPS流量，但是在开启Netflix分流服务器以后，Emby和Plex获得的广域网IP是VPS的IP地址，造成外部网无法访问Plex服务器。
> > 自己找过解决办法，就是关闭Netflix分流服务器，一切正常。也尝试在Netflix分流的IP里没有找到和Plex相关的域名和IP，不知道为何只要开启Netflix分流服务器，Plex和Emby就无法获取本地公网IP。
> 
> 我觉得这不是问题，代理服务都是正向的，也就是你群晖常用端口的出口流量才会走代理（ssrp客户端页面可以设置‘仅常用端口（不走p2p流量到代理）’），所以PT上传\下载不会误伤VPS流量。你群晖获取的ip地址是VPS地址这也是正常的，因为一般都是curl请求获取公网地址，走的80/443端口属于常用端口，常用端口默认走代理了～

非常感谢您的回答，但请听我说：第一，我把群晖的IP设置了仅常用端口；第二，我也把群晖的IP设置了不走代理服务器，只要开了分流服务器，以上功能全部失效，我是想反馈这个问题。但如果没有开分流服务器，那么仅常用端口这一步可以省略，直接设置群晖的IP不走代理，这样PT、Plex、Emby都可以获取到本地家庭宽带的IP

**Comment by qiqiandfei** [comment]
> > > 家里有群晖，除了日常功能以外，还有PT下载，Plex服务器，Emby服务器。已经把群晖的内网IP设置变成了不走代理流量，避免PT下载误伤VPS流量，但是在开启Netflix分流服务器以后，Emby和Plex获得的广域网IP是VPS的IP地址，造成外部网无法访问Plex服务器。
> > > 自己找过解决办法，就是关闭Netflix分流服务器，一切正常。也尝试在Netflix分流的IP里没有找到和Plex相关的域名和IP，不知道为何只要开启Netflix分流服务器，Plex和Emby就无法获取本地公网IP。
> > 
> > 
> > 我觉得这不是问题，代理服务都是正向的，也就是你群晖常用端口的出口流量才会走代理（ssrp客户端页面可以设置‘仅常用端口（不走p2p流量到代理）’），所以PT上传\下载不会误伤VPS流量。你群晖获取的ip地址是VPS地址这也是正常的，因为一般都是curl请求获取公网地址，走的80/443端口属于常用端口，常用端口默认走代理了～
> 
> 非常感谢您的回答，但请听我说：第一，我已经设置了仅常用端口；第二，我同时也把群晖的IP设置了不走代理服务器，只要开了分流服务器，以上功能全部失效，我是想反馈这个问题。但如果没有开分流服务器，那么仅常用端口这一步可以省略，直接设置群晖的IP不走代理，这样PT、Plex、Emby都可以获取到本地家庭宽带的IP

首先您说的这个问题确实存在（之前的版本），我的Switch在开了Netflix分流后获取到的地址是Netflix的分流节点地址。即使是把Switch加到不走代理也是不生效的。我上周刚刚升级了系统，升级后还没来得及测试，改天我抽空测试一下，看看这个问题是否还存在。
其次，群晖为什么一定要获取到本地家庭宽带的IP地址呢？是不是可以通过路由器添加端口转发解决？路由器拨号，得到公网地址，弄个ddns绑个域名是不是就能解决您的问题了？也就是您从公网访问Plex、Emby都是先到路由再到群晖，毕竟只有路由是暴露在公网的～

**Comment by jackiezhang29** [comment]
> > > > 家里有群晖，除了日常功能以外，还有PT下载，Plex服务器，Emby服务器。已经把群晖的内网IP设置变成了不走代理流量，避免PT下载误伤VPS流量，但是在开启Netflix分流服务器以后，Emby和Plex获得的广域网IP是VPS的IP地址，造成外部网无法访问Plex服务器。
> > > > 自己找过解决办法，就是关闭Netflix分流服务器，一切正常。也尝试在Netflix分流的IP里没有找到和Plex相关的域名和IP，不知道为何只要开启Netflix分流服务器，Plex和Emby就无法获取本地公网IP。
> > > 
> > > 
> > > 我觉得这不是问题，代理服务都是正向的，也就是你群晖常用端口的出口流量才会走代理（ssrp客户端页面可以设置‘仅常用端口（不走p2p流量到代理）’），所以PT上传\下载不会误伤VPS流量。你群晖获取的ip地址是VPS地址这也是正常的，因为一般都是curl请求获取公网地址，走的80/443端口属于常用端口，常用端口默认走代理了～
> > 
> > 
> > 非常感谢您的回答，但请听我说：第一，我已经设置了仅常用端口；第二，我同时也把群晖的IP设置了不走代理服务器，只要开了分流服务器，以上功能全部失效，我是想反馈这个问题。但如果没有开分流服务器，那么仅常用端口这一步可以省略，直接设置群晖的IP不走代理，这样PT、Plex、Emby都可以获取到本地家庭宽带的IP
> 
> 首先您说的这个问题确实存在（之前的版本），我的Switch在开了Netflix分流后获取到的地址是Netflix的分流节点地址。即使是把Switch加到不走代理也是不生效的。我上周刚刚升级了系统，升级后还没来得及测试，改天我抽空测试一下，看看这个问题是否还存在。
> 其次，群晖为什么一定要获取到本地家庭宽带的IP地址呢？是不是可以通过路由器添加端口转发解决？路由器拨号，得到公网地址，弄个ddns绑个域名是不是就能解决您的问题了？也就是您从公网访问Plex、Emby都是先到路由再到群晖，毕竟只有路由是暴露在公网的～

可能我的表述有误，我所说的本地家庭宽带IP是外网IP，不是内网IP，其次，从公网访问Plex和Emby，一定要让群晖获得家宽外网IP才可以，如果是VPS的IP，Plex就无法从公网访问

**Comment by qiqiandfei** [comment]
是的 我没有看错 公网ip在路由上也可以实现你的需求 需要路由做一次端口转发到你的群晖就行了！群晖本身不需要外网ip，只有内网ip一样可以从公网访问，只要群晖和你的路由在同一个局域网下～

**Comment by jackiezhang29** [comment]
你说的这些我都知道，现在的问题是，开了分流，群晖上的Plex服务器获取的就是VPS的iP，造成了公网客户端无法访问

**Comment by qiqiandfei** [comment]
我昨天回家试过了，新版本貌似没有这个问题了，把Switch放到"不走代理的局域网LAN IP",在开启Netflix分流时，Switch获取到的地址是路由器的公网地址～

**Comment by jackiezhang29** [comment]
> 我昨天回家试过了，新版本貌似没有这个问题了，把Switch放到"不走代理的局域网LAN IP",在开启Netflix分流时，Switch获取到的地址是路由器的公网地址～

我看了你的回复，也更新了版本，貌似的确是好了，多谢提供的帮忙，我暂时不关闭，等测试几天再看
